api/hypershift/v1beta1/hostedcluster_types.go

```diff
@@ -1859,6 +1859,97 @@ type AzurePlatformSpec struct {
 	// +immutable
 	// +required
 	SecurityGroupID string `json:"securityGroupID,omitempty"`
+
+	// ManagedIdentities contains the client IDs related to the managed identities needed for HCP control plane
+	// and data plane components that authenticate with Azure's API.
+	//
+	// +kubebuilder:validation:Required
+	// +required
+	ManagedIdentities *AzureResourceManagedIdentities `json:"managedIdentities,omitempty"`
+}
+
+// AzureResourceManagedIdentities contains the client IDs related to the managed identities needed for HCP control plane
+// and data plane components that authenticate with Azure's API.
+type AzureResourceManagedIdentities struct {
+	// ControlPlaneMIs contains the client IDs of all the managed identities on the HCP control plane needing to
+	// authenticate with Azure's API.
+	//
+	// +kubebuilder:validation:Required
+	// +required
+	ControlPlaneMIs *ControlPlaneManagedIdentities `json:"controlPlaneMIs"`
```

Comment:
Required structs should not be pointers
```suggestion
	ControlPlaneMIs ControlPlaneManagedIdentities `json:"controlPlaneMIs"`
```